obs-studio-server/source/nodeobs_api.cpp

```diff
@@ -717,6 +727,34 @@ void writeCrashHandler(std::vector<char> buffer)
 	close(file_descriptor);
 }
 #endif
+
+static bool checkIfDebugLogsEnabled(const std::string& appdata)
+{
+#if defined(_DEBUG)
+	return true;
+#else
+	// When you change the environment variable and start
+	// Streamlabs Desktop via a console/terminal,
+	// you may have to restart the console/terminal.
+	// Using the environment variable on macOS may be even more complicated:
+	// https://stackoverflow.com/questions/135688/setting-environment-variables-on-os-x
+	char* envValue = getenv("SL_DESKTOP_ENABLE_DEBUG_LOGS");
+	if (envValue != nullptr) {
+		if (astrcmpi(envValue, "on") == 0 || astrcmpi(envValue, "yes") == 0) {
+			return true;
+		}
+	}
+	// Even if the environment variable is set and the logs are turned off,
+	// we still check for the file to avoid issues with finding where the variable is defined.
+	const std::string filename = appdata + "/enable-debug-logs";
```

Comment:
Why do the debug logs need to go to a separate file when it's enabled? Could we not still use the same log file?

Reply:
This is not a log file. This is a file to check if debug logs are enable. I would like to have both ways (the environment variable and the file) because most likely on macOS the environment variable way is not convenient.